src/modules/actions/summariseByAI.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { forwardRef } from 'react'
+
+import flag from 'cozy-flags'
+import ActionsMenuItem from 'cozy-ui/transpiled/react/ActionsMenu/ActionsMenuItem'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import MagicTrickIcon from 'cozy-ui/transpiled/react/Icons/MagicTrick'
+import ListItemIcon from 'cozy-ui/transpiled/react/ListItemIcon'
+import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
+import { isFileSummaryCompatible } from 'cozy-viewer/dist/helpers'
+
+const makeComponent = (label, icon) => {
+  const Component = forwardRef((props, ref) => {
+    return (
+      <ActionsMenuItem {...props} ref={ref}>
+        <ListItemIcon>
+          <Icon icon={icon} />
+        </ListItemIcon>
+        <ListItemText primary={label} />
+      </ActionsMenuItem>
+    )
+  })
+
+  Component.displayName = 'summariseByAI'
+
+  return Component
+}
+
+export const summariseByAI = ({ t, hasWriteAccess, navigate }) => {
+  const label = t('actions.summariseByAI')
+  const icon = MagicTrickIcon
+
+  return {
+    name: 'summariseByAI',
+    label,
+    icon,
+    displayCondition: files =>
+      flag('ai.available') &&
+      isFileSummaryCompatible(files[0]) &&
+      hasWriteAccess,
+    action: files => {
+      const file = files[0]
+      navigate(`file/${file._id}`, {
+        state: { showAIAssistant: true }
+      })
```

Comment:
Interesting, so this allow to open directly the viewer to the AI panel and if you click back it goes to file list and not file viewer?

Reply:
Oh I see that a lot of comments have disappeared from Figma 😭 I asked Swann and he answered that yes you should go back to file list.

So maybe no need for routing API ?